### PR TITLE
PBI-99-117: JWT Service

### PR DESCRIPTION
**Pull request by ferryhieronimus**
## Describe your changes

This PR introduces a JWT service for token management. Note that in the tests, we do not test the JWT package itself but focus on our own implementation. Currently we use UserID for the Subject but this is subject to change (no pun intended)

## Further Improvement

Currently, the Claims struct is tightly coupled with JWT. In the future, this could be generalized, possibly using adapter pattern.

## Issue ticket number and link

ADU-99: https://linear.app/adudu/issue/ADU-99/jwt-generation-service
ADU-117: https://linear.app/adudu/issue/ADU-117/[unit-test]-jwt-service

### CODE REVIEW

File: internal/token/jwt.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	if s.cfg.Secret == "" {
+		return "", errors.New("secret key is empty")
+	}
+
+	tokenObject := jwt.NewWithClaims(jwt.SigningMethodHS256, Claims{
+		RegisteredClaims: jwt.RegisteredClaims{
+			Subject:   spec.UserID,
+			ExpiresAt: jwt.NewNumericDate(s.clock.Now().UTC().Add(30 * 24 * time.Hour)), // one month from now
+			IssuedAt:  jwt.NewNumericDate(s.clock.Now().UTC()),
+			ID:        uuid.NewString(),
```
Comment (edutjie) [inline internal/token/jwt.go:28]: Is this ID related to Account ID or no? If it is, we can't use uuid since the ID in Account db table only support varchar(15).

File: internal/token/jwt.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+type jwtManager struct {
+	cfg   config.Token
+	clock clock.Clock
+}
+
+func (s *jwtManager) GenerateToken(spec ClaimSpec) (string, error) {
+	if s.cfg.Secret == "" {
+		return "", errors.New("secret key is empty")
+	}
```
Comment (valeelim) [inline internal/token/jwt.go:21]: I think the `validate` on the config declaration already handles this

File: internal/token/jwt_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+		jwtMgr = &jwtManager{
+			cfg:   config.Token{Secret: "secret"},
+			clock: mockClock,
+		}
+
+		return gomega.NewWithT(t)
+	}
+
+	t.Run("GenerateTokenWithValidClaimsReturnsTokenSuccessfully", func(t *testing.T) {
```
Comment (valeelim) [inline internal/token/jwt_test.go:36]: nit: since the test is wrapped in `Test_GenerateToken`, we could rename this to `valid claim returns successfully`

File: internal/token/jwt.go (hunk 1 above)
Comment (ferryhieronimus) [reply internal/token/jwt.go:28]: No. This is ID for JWT

Comment (valenciusap17) [inline internal/token/jwt.go:28]: Could you explain to me why you put an ID inside JWT token? It is actually my first time seeing this case

Comment (ferryhieronimus) [reply internal/token/jwt.go:28]: ID is "jti" in JWT. It can be used to prevent the JWT from being replayed. [This is](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.7) a great read for you!

File: internal/token/jwt.go (hunk 2 above)
Comment (ferryhieronimus) [reply internal/token/jwt.go:21]: This is for double protection i guess
